src/mpi/request/request_impl.c

```diff
@@ -301,31 +301,13 @@ int MPIR_Test_impl(MPIR_Request * request_ptr, int *flag, MPI_Status * status)
     return MPIR_Test_state(request_ptr, flag, status, NULL);
 }
 
-int MPIR_Test(MPI_Request * request, int *flag, MPI_Status * status)
+int MPIR_Test(MPIR_Request * request_ptr, int *flag, MPI_Status * status)
 {
     int mpi_errno = MPI_SUCCESS;
-    MPIR_Request *request_ptr = NULL;
-
-    MPIR_Request_get_ptr(*request, request_ptr);
-    MPIR_Assert(request_ptr != NULL);
 
     mpi_errno = MPID_Test(request_ptr, flag, status);
     MPIR_ERR_CHECK(mpi_errno);
```

Comment:
if we go this route we can remove `MPIR_Test` entirely and just call the device API from the binding layer

Reply:
I pulled this thread a little more and removed the MPIR functions along with some other changes. Open to feedback.